### ISSUE TITLE
A generic linter to ease building new ones

### ISSUE THREAD

**Issue by fgarcia** [w0rp/ale#1011]
I am trying to build a custom linter, but I am not very good with VimL.

It would be great having a generic linter with an standard / documented input. 

With that feature, I could easily write a custom linter in my best language. The objective would be capturing the output of one (or many) tools and print a single report that the VimL generic linter understand.

I would only need to know the message format (and useful fields) that ALE supports. 

The VimL linter could be very simple, but the custom linters can go from basic (plain stdout adapter) to crazy complex like querying web services.

**Comment by w0rp** [comment]
I'm not entirely sure what you're asking for. Do you want someone to build a tool which somehow wraps other tools? Isn't that what ALE does already?

**Comment by w0rp** [comment]
I think you're asking for someone to write a program which works as a middle-man between ALE and other programs. That would be pretty pointless. That would achieve exactly what ALE does already, but with another layer of abstraction, and then you'd have to solve all of the same problems ALE solves already, but in a different place.

I think I'll say "no" to this.

**Comment by fgarcia** [comment]
The core idea is **enable non VimL programmers to extend ALE**. Currently you are responsible of maintaining each linter adapter in this repository. If a non VimL programmer wants a new linter in Vim - ALE their only way is opening an issue or learning VimL

However if there is a linter that takes an basic input like

` [error/warning] : [file path] : [line] : [column]`

Now non VimL people could implement a lint adapter in any language. They just have to write a command line tool that prints that output

So basically the problem is this: Do you want non VimL coders to depend on you for having a new linter or modify an existing one?

For those who know VimL this might sounds quite useless. However for me having a way to prototype / explore linter ideas without learning VimL and without depending on an issue to be accepted can change everything.

**Comment by w0rp** [comment]
No, learn VimL.

**Comment by w0rp** [comment]
Many different users have contributed linters. The vast majority of linters were written by many different contributers, and they have done a great job. If you want to contribute to a project, you must learn the language the project is written in.

It's not reasonable to ask for part of a project to be written in some other language because it might somehow attract other people. VimL is the appropriate tool for the job, and it's not hard to learn. If you want to contribute, learn to write VimL.
